packages/jss/src/PluginsRegistry.js.flow

```diff
@@ -0,0 +1,41 @@
+﻿// @flow
+import type {
+  Plugin,
+  Rule,
+  RuleOptions,
+  UpdateOptions,
+  JssStyle,
+  JssValue,
+  OnCreateRule,
+  OnProcessRule,
+  OnProcessStyle,
+  OnProcessSheet,
+  OnChangeValue,
+  OnUpdate
+} from './types'
+import type StyleSheet from './StyleSheet'
+import type {StyleRule, BaseStyleRule} from './plugins/styleRule'
+
+type Registry = {
+  onCreateRule: Array<OnCreateRule>,
+  onProcessRule: Array<OnProcessRule>,
+  onProcessStyle: Array<OnProcessStyle>,
+  onProcessSheet: Array<OnProcessSheet>,
+  onChangeValue: Array<OnChangeValue>,
+  onUpdate: Array<OnUpdate>
+}
+
+declare export default class PluginsRegistry {
+  plugins: {|
```

Comment:
Would we even need to declare this as I wouldn't consider this to be a "public" API and shouldn't really be accessed from the outside?

Reply:
yeah, was my question as well, since typings are only used from outside,  I am not sure internally used types provide any additional type safety, unless used from outside, but thats not going to happen

Reply:
I think it should be removed as long as we only want to use these types for the outside and not for internal type checking

Reply:
Just tried, can't remove completely all private fields from .flow, it's still gonna be a mixture of public apis and internal once, otherwise it won't compile

Reply:
Because some fields are accessed by jss core itself, from other modules, for actual public api we would need to differentiate that and the access from public apis